andela/travel_tool_front
# #159725850 Consume api to post a new request

#### What does this PR do?
Consumes api to post a new request

#### Description of Task to be completed?
- create handleChange and handleSubmit methods
- create actions to create a new request
- create reducer to create a new request
- write tests for action, reducers, services and sagas
- create saga middleware to make a post request to the server
- set token globally so that it can be available on every api call
- add requestSaga tot the root saga

#### How should this be manually tested?
- clone the repo
- run `cd travel-tool-front` to switch the directory to the repo
- checkout to this branch  with `git checkout ft-consume-create-requests-api-159725850`
- install dependencies with `yarn`
- start the docker engine, then run `make start` to start the server
- if you want to connect to the backend api locally setup the backend 
- supply the server side of the app by following setup instructions in the `README` section of the repo [server side repo](https://github.com/andela/travel_tool_front/pull/43)
- start the backend server
- setup environmental variables for the client and server side
- navigate to `localhost:3000` in a browser
- login with your Andela email to gain access to the requests page
- on the requests page, click the `New Request` button
- fill the modal with the required information and submit your travel request
- if your request is successful, you should see a message notifying you

#### Any background context you want to provide?
Depends on the `add request modal feature`- [159725850](https://www.pivotaltracker.com/story/show/159725850)

To fully test this feature, it might be important to setup the server side of the app `locally`, the following steps guide on how to setup the backend
- clone the backend repo [back end repo](https://github.com/andela/travel_tool_front/pull/43)
- run `yarn` to install dependencies
- setup the env variables by following the guide in `.env.example`
the `env` variables include a `JWT_PUBLIC_KEY` which can be gotten from the app's admin
- run `yarn db:migrate` to trun database migrations
- run `yarn start:dev` to start the server

#### Screenshots(if possible)

![image](https://user-images.githubusercontent.com/25783086/44549356-be62cd80-a718-11e8-8dc4-f6ffa8c27d8d.png)

![image](https://user-images.githubusercontent.com/25783086/44549327-a8eda380-a718-11e8-9527-4b71441795da.png)
#### What are the relevant pivotal tracker stories?
[159725850](https://www.pivotaltracker.com/story/show/159724397)
